incubator/hnc/internal/reconcilers/hnc_config.go

```diff
@@ -437,57 +442,61 @@ func (r *ConfigReconciler) setTypeStatuses(inst *api.HNCConfiguration) {
 	inst.Status.Types = statuses
 }
 
-// enqueueSingleton enqueues the `config` singleton to trigger the reconciliation
-// of the singleton for a given reason . This occurs in a goroutine so the
-// caller doesn't block; since the reconciler is never garbage-collected,
-// this is safe.
-func (r *ConfigReconciler) enqueueSingleton(log logr.Logger, reason string) {
-	go func() {
-		log.Info("Enqueuing for reconciliation", "reason", reason)
-		// The watch handler doesn't care about anything except the metadata.
-		inst := &api.HNCConfiguration{}
-		inst.ObjectMeta.Name = api.HNCConfigSingleton
-		r.Trigger <- event.GenericEvent{Meta: inst}
-	}()
+// RequestReconcile records that the reconciler needs to be reinvoked.
+func (r *ConfigReconciler) RequestReconcile(reason string) {
```

Comment:
Do we need to make this method exported? Are we expecting this to be called outside the package?

Reply:
(As discussed in another comment) no and no :) This is mainly about saying that this method _should_ be called outside of the ConfigReconciler.